src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/DiagnosticCounter.cs

```diff
@@ -125,12 +125,32 @@ internal string GetMetadataString()
                 return "";
             }
 
-            StringBuilder sb = new StringBuilder("");
-            foreach(KeyValuePair<string, string> kvPair in _metadata)
+            // The dictionary is only initialized to non-null when there's metadata to add, and no items
+            // are ever removed, so if the dictionary is non-null, there must also be at least one element.
+            Dictionary<string, string>.Enumerator enumerator = _metadata.GetEnumerator();
+            Debug.Assert(_metadata.Count > 0);
+            bool gotOne = enumerator.MoveNext();
+            Debug.Assert(gotOne);
```

Comment:
how could count be > 0 on a dictionary but the first MoveNext on its enumerator fail?

Reply:
That's why it's asserted rather than checked with an if.

Reply:
I guess I'm saying is a test that dictionary isn't broken, which isn't normally something we test elsewhere.

Reply:
It's a test that the condition holds. There are various things that could influence it, e.g. another thread removing the item after Count returns 1, the side code being refactored and the count check getting removed. Asserts exist o validate things that should be guaranteed. It also documents for a reader that we're choosing to explicitly ignore the Boolean result of MoveNext, which is unusual. You think the assert has no benefit and should be removed?